py-deploy/tldeploy/identity.py

```diff
@@ -267,3 +271,55 @@ def filled_and_signed_meta_transaction(
 
     def get_next_nonce(self):
         return self.contract.functions.lastNonce().call() + 1
+
+
+def deploy_proxied_identity(web3, factory_address, implementation_address, signature):
+    owner = recover_proxy_deployment_signature_owner(
+        web3, factory_address, implementation_address, signature
+    )
+
+    proxy_interface = get_contract_interface("Proxy")
+    initcode = build_initcode(
+        contract_abi=proxy_interface["abi"],
+        contract_bytecode=proxy_interface["bytecode"],
+        constructor_args=[owner],
+    )
+
+    factory_interface = get_contract_interface("IdentityProxyFactory")
+    factory = web3.eth.contract(
+        address=factory_address,
+        abi=factory_interface["abi"],
+        bytecode=factory_interface["bytecode"],
+    )
+
+    tx_id = factory.functions.deployProxy(
+        initcode, implementation_address, signature
+    ).transact({"from": web3.eth.accounts[0]})
+    wait_for_successful_transaction_receipt(web3, tx_id)
+
+    proxy_address = build_create2_address(factory_address, initcode)
```

Comment:
Can we not better ask for the address from the deployment transaction. We have events for it right? This way it is easier to spot an issue if there is any I think

Reply:
Yes, we can use the events